packages/sit-onyx/src/components/TestInput/TestInput.vue

```diff
@@ -71,29 +73,43 @@ const emit = defineEmits<{
   validityChange: [state: ValidityState];
 }>();
 
-const isTouched = ref(false);
+const { t } = injectI18n();
 
-const inputElement = ref<HTMLInputElement | null>(null);
+const value = computed({
+  get: () => props.modelValue,
+  set: (value) => emit("update:modelValue", value),
+});
 
+const isTouched = ref(false);
+const inputElement = ref<HTMLInputElement | null>(null);
 const validityState = ref(inputElement.value?.validity);
 
 const errorMessage = computed(() => {
-  /* when the validity state is uninitialized or the form is valid, we don't show an error. */
   if (!validityState.value || validityState.value.valid) return "";
-  /* a custom error message always is considered first */
-  if (props.errorMessage) return props.errorMessage;
-
-  const element = inputElement.value;
-  /* when a language key is provided, we offer our own translations of the error messages 
-  to match the rest of the user's application */
-  if (props.lang && element) return useFormValidationMessage(element.validity, props);
-  /* we default to the browser's standard validation message that relies on the browser language */
-  return element?.validationMessage || "";
-});
 
-const value = computed({
-  get: () => props.modelValue,
-  set: (value) => emit("update:modelValue", value),
+  const errorType = getFirstInvalidType(validityState.value);
+  if (!errorType) return "";
+
+  // a custom error message always is considered first
+  if (props.errorMessage || errorType === "customError") return props.errorMessage;
```

Comment:
to honor the comment's message, I'd actually put that first in the code aswell ;)
if it's invalid, the actual active error type should not matter. also, customError will only ever be possible when the projects would be able to call the [setCustomValidity](https://developer.mozilla.org/en-US/docs/Web/API/HTMLObjectElement/setCustomValidity) function on the inner input itself, is that even possible with our new component structure?
Additionally, setCustomValidity is messy because it immediately sets valid to false (side effects) and the projects have to reset it by hand. that's why I offered preferably just an error message override option. 
Also this should allow the projects to also ajdust something like a patternMismatch situationally.


```suggestion
  // a custom error message always is considered first
  if (props.errorMessage) return props.errorMessage;
  
  const errorType = getFirstInvalidType(validityState.value);
  if (!errorType) return "";
```

Reply:
if you want we can have a call to clarfiy this topic :)

Reply:
That will give a TypeScript issue since the errorType of the ValidityState can be "customError" where we don't have a translation so I am returning the custom error message in this case

Reply:
then I'd rather return an empty string as the existence of customError does not mean that the projects have passed over a props.errorMessage

Reply:
There are two approaches
1) we watch the props.errorMessage and call setCustomValidity() with the value -> when the errorMessage is reset, we would automatically reset the error state when calling setCustomValidity with an empty string :)
But the down side is, that the previous workflow does not work anymore which I showcased in the FormDemo component. the workflow was
- the input got a pattern (lowercase characters only) assigned
- when the user typed in an uppercase character, the validityChange Event fired and the patternMismatch was true
- the dev was able to detect that and set customErrorMessage that explained which pattern is expected, unlike our generic standard "your pattern is not ok" message
- now with the above behavior, it would kill itself - when setting errorMessage, the TestInput fires setCustomValidity("You need to use lowercase only") and this fires the validityChanged event again. our condition in the FormElement now sees that patternMismatch is not the active error anymore and resets the errorMessage?
I guess we would need to check if patternMismatch is still kept even though customError is set. if that's the case, no problem exists and we could implement it like that.
Should I try it out on your branch with my FormElement component from my older branch?

2) could we offer setCustomValidity as a method that can be called on the TestInput component?

Reply:
Should we delay this feature then? I think its out of scope of this PR. Maybe we should remove the custom error message for now and add it later when we defined a concept for it? What do you think?

Reply:
created follow-up ticket #113